nef/Preferences/PreferencesView.swift

```diff
@@ -0,0 +1,114 @@
+                CheckOptionView(text: i18n.Description.showWatermark, nested: true, selection: $viewModel.showWatermark)
+            }.padding(.all).offset(x: -12)
+
+            VStack {
+                PickerOptionView(title: i18n.Option.font, items: viewModel.fontItems, selection: $viewModel.selectionFont)
+                PickerOptionView(title: i18n.Option.theme, items: viewModel.themeItems, selection: $viewModel.selectionTheme)
+                PickerOptionView(title: i18n.Option.size, items: viewModel.sizeItems, selection: $viewModel.selectionSize)
+                PickerOptionView(title: i18n.Option.color, items: viewModel.colorItems, selection: $viewModel.selectionColor)
+                ColorOptionView(value: $viewModel.hex)
+            }.padding(.bottom).offset(x: -12)
+            
+            // Carbon viewer
+            CarbonViewer(state: $viewModel.state).frame(maxWidth: .infinity)
+                .cornerRadius(10, antialiased: true)
+                .padding(20)
+            
+        }.border(NefColor.gray)
+         .background(NefColor.white)
+         .frame(maxWidth: .infinity, maxHeight: .infinity)
+         .padding(20)
+         .onAppear(perform: onAppear)
+    }
+    
+    // MARK: private methods
+    private func onAppear() {
+        viewModel.onAppear()
+    }
+    
+    private func restore() {
+        viewModel.tapOnRestore()
+    }
+    
+    // MARK: - Constants
+    enum i18n {
+        static let title = NSLocalizedString("Carbon", comment: "")
+        
+        enum Option {
+            static let font = NSLocalizedString("Font type", comment: "")
+            static let theme = NSLocalizedString("Theme", comment: "")
+            static let color = NSLocalizedString("Background color", comment: "")
+            static let size = NSLocalizedString("Size", comment: "")
+        }
+        
+        enum Description {
+            static let showLines = NSLocalizedString("Line numbers", comment: "")
+            static let showWatermark = NSLocalizedString("Watermark", comment: "")
+        }
+    }
+    
+    enum Layout {
+        static let leftPanel: CGFloat  = 150
+        static let rightPanel: CGFloat = 220
+    }
+    
+    enum NefImage {
+        static let restore = Image("restore")
+    }
+    
+    enum NefColor {
+        static let gray = Color(red: 0.78, green: 0.78, blue: 0.78)
+        static let white = Color(red: 1, green: 1, blue: 1)
+    }
+}
+
+// MARK: - CarbonStyle View
+import NefModels
+
+extension CarbonStyle.Font {
+    var itemName: String { rawValue }
+}
+
+extension CarbonStyle.Size {
+    var itemName: String { "\(rawValue)".replacingOccurrences(of: ".0", with: "x") }
+}
+
+extension CarbonStyle.Theme {
+    var itemName: String { rawValue.replacingOccurrences(of: "-", with: " ").capitalized }
+}
+
+extension String {
+    var itemColorName: String { self != "nef" ? capitalized : self }
```

Comment:
This feels a bit strange as an extension of String; it is particular to how you render the names of the color list and should be a function somewhere else.

Reply:
I do agree. To give context, I will add in `CarbonStyle` as a static method.

```
extension CarbonStyle {
    static func itemColorName(in value: String) -> String
}
```

Reply:
I will keep on the same place, because this View will render these options, and it should define how to display each element.

Example of use:
```swift
colors.keys.sorted().map(CarbonStyle.itemColorName)
```

Reply:
LGTM!